README.md

```diff
@@ -38,26 +37,79 @@ SELECT hash, author_email, author_name FROM commits LIMIT 2;
+| remotes      |repository_id, name, push_url,fetch_url,push_refspec,fetch_refspec                                   | 
+|    commits   | hash, author_name, author_email, author_when, comitter_name, comitter_email, comitter_when, message, tree_hash |
+|     blobs    | hash, size, content                                                                                 |
+|  refs        | repository_id, name, hash                                                                           |
 | tree_entries | tree_hash, entry_hash, mode, name                                                                   |
 
-## SQL syntax
+## Functions
+
+To make some common tasks easier for the user, there are some functions to interact with the previous mentioned tables:
+
+|     Name     |                                               Description                                           |
+|:------------:|:---------------------------------------------------------------------------------------------------:|
+|commit_has_blob(commit_hash,blob_hash)bool| get if the specified commit contains the specified blob                 |
+|history_idx(start_hash, target_hash)int| get the index of a commit in the history of another commit                 |
+|is_remote(reference_name)bool| check if the given reference name is from a remote one                               |
+|is_tag(reference_name)bool| check if the given reference name is a tag                                              |
 
-We are continuously adding more functionality to gitquery. We support a subset of the SQL standard, currently including:
+## Examples
+
+### Get all the HEAD references from all the repositories
+```sql
+SELECT * FROM refs WHERE name = 'HEAD'
+
+```
 
-|                        |                                     Supported                                     |
-|:----------------------:|:---------------------------------------------------------------------------------:|
-| Comparison expressions |                                !=, ==, >, <, >=,<=                                |
-|  Grouping expressions  |                                    COUNT, FIRST                                   |
-|  Standard expressions  |                              ALIAS, LITERAL, STAR (*)                             |
-|       Statements       | CROSS JOIN, DESCRIBE, FILTER (WHERE), GROUP BY, LIMIT, SELECT, SHOW TABLES, SORT  |
+### Commits that appears in more than one reference
+
+```sql
+SELECT * FROM (
+				SELECT COUNT(c.hash) AS num, c.hash
+				FROM refs r
+				INNER JOIN commits c
+					ON history_idx(r.hash, c.hash) >= 0
+				GROUP BY c.hash
+			) t WHERE num > 1
+```
+
+###  Get the number of blobs per HEAD commit
+```sql
+SELECT COUNT(c.hash), c.hash
+			FROM refs r
+			INNER JOIN commits c
+				ON history_idx(r.hash, c.hash) >= 0
+			INNER JOIN blobs b
+				ON commit_has_blob(c.hash, b.hash)
+			WHERE r.name = 'HEAD'
+			GROUP BY c.hash
```

Comment:
move this filter to the ON like `ON r.name = 'HEAD' AND history_idx`.  The performance improvement is huge because history_idx needs to be called way less times.